### PR TITLE
Improved CPU performance

### PR DESCRIPTION
**Pull request by smitdylan2001**
### Purpose of this PR

Improved CPU performance using:

- Use native Lenth/Count instead of LINQ
- Re-order vector calculations to be faster
- merge .rotation and .position into .SetPositionAndRotation
- Use better dictionary usage
- Use TryGetComponent for less garbage allocation
- Merge LINQ statements
- Merge if/else return into single return
- Use OfType instead of Where().Cast
- Use Color32 (in UVEditor.cs, please double check)
- Use AddRange instead of looped Add

### Comments to Reviewers

The Color32 improvement in UVEditor worked for me, but since this is the only major change this should be double checked

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.probuilder?pullRequest=445) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.probuilder?pullRequest=445) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.probuilder?pullRequest=445) it.</sub>

File: Runtime/Core/ShapeGenerator.cs [hunk 1]
```diff
@@ -1562,7 +1562,7 @@ public static ProBuilderMesh GenerateTorus(PivotLocation pivotType, int rows, in
 
             int col = clampedColumns - 1;
 
-            Vector3[] cir = GetCirclePoints(clampedRows, clampedTubeRadius, clampedVerticalCircumference, Quaternion.Euler(Vector3.up * 0f * clampedHorizontalCircumference), clampedRadius);
+            Vector3[] cir = GetCirclePoints(clampedRows, clampedTubeRadius, clampedVerticalCircumference, Quaternion.Euler(0f * clampedHorizontalCircumference * Vector3.up), clampedRadius);
```
Comment (karl-) [inline Runtime/Core/ShapeGenerator.cs:1565]: I understand that this is just changing the order of operations to multiply single values before vec3, but why are we multiplying `0f * clampedHorizontalCircumference * Vector3.up` in the first place? Isn't this equivalent to simply writing `vec3(0,0,0)`?

**Review by karl- — APPROVED** [review]

File: Runtime/Core/ShapeGenerator.cs (hunk 1 above)
Comment (smitdylan2001) [reply Runtime/Core/ShapeGenerator.cs]: Good question and no idea how I overlooked this (or why it was there in the first place). Fixed in new commit

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - [case: PBLD-34] Fixed a bug where `Experimental Features Enabled` was not activating when using `Dedicated Server` platform. 
 - [case: PBLD-32] Fixed `New Shape` start location being incorrect after using right mouse button.
 - [case: PBLD-19] Fixed shape creation when the camera perspective is set to Top.
+- Improved CPU performance
```
[the remark this answers is not in the document]
Comment (smitdylan2001) [reply CHANGELOG.md]: Agreed that it is the better comment

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - [case: PBLD-34] Fixed a bug where `Experimental Features Enabled` was not activating when using `Dedicated Server` platform. 
 - [case: PBLD-32] Fixed `New Shape` start location being incorrect after using right mouse button.
 - [case: PBLD-19] Fixed shape creation when the camera perspective is set to Top.
+- Minor performance improvements and reduced allocations when querying for components.
```
Comment (JonMUnity) [inline CHANGELOG.md:15]: Needs to start with a verb. 
```suggestion
- Made minor performance improvements and reduced allocations when querying for components.
```

Comment (smitdylan2001) [reply CHANGELOG.md]: Got it

**Comment by smitdylan2001** [comment]
Any chance of this getting merged?

**Review by JonMUnity — APPROVED** [review]

**Pull request merged**